### PR TITLE
fix(core): bringing back lazy behavior for combine

### PR DESCRIPTION
**Pull request by Fyzu**
- Bringing back lazy behavior for combine
- Distinct changes in interop

### CODE REVIEW

File: packages/lens/src/lensed-atom.spec.ts [hunk 1]
```diff
@@ -109,28 +109,4 @@ describe('view', () => {
 		expect(ao.next).toHaveBeenCalledTimes(1)
 		expect(bo.next).toHaveBeenCalledTimes(1)
 	})
-	it('distinct changes', () => {
```
Comment (Fyzu) [inline packages/lens/src/lensed-atom.spec.ts:112]: Not suitable, coz combine now emits even without changing the value

File: packages/core/src/property.spec.ts [hunk 2]
```diff
@@ -156,26 +156,6 @@ describe('combine', () => {
 		expect(cb).toHaveBeenCalledTimes(1)
 		expect(cb).toHaveBeenLastCalledWith(2)
 	})
-	it('does not emit if projected value did not change and combined property has at least one consumer', () => {
```
Comment (Fyzu) [inline packages/core/src/property.spec.ts:159]: Not suitable, coz combine now emits even without changing the value

File: packages/core/src/interop-observable.ts [hunk 3]
```diff
@@ -25,9 +25,14 @@ export const newInteropObservable = <A>(
 	subscribe: Observable<Time>['subscribe'],
 ): InteropObservable<A> => ({
 	subscribe: (observer) => {
+		let value: A | undefined
 		const emitNext = () => {
 			if (observer.next) {
-				observer.next(get())
+				const prevValue = value
+				value = get()
+				if (prevValue !== value) {
```
Comment (raveclassic) [inline packages/core/src/interop-observable.ts:33]: This breaks if the first value supplied to interop observable is `undefined`.

File: packages/core/src/property.spec.ts [hunk 4]
```diff
@@ -192,32 +172,20 @@ describe('combine', () => {
...
 		c.subscribe({ next: cb1 })
-		c.subscribe({ next: cb2 })
-		// imitate consumer to warm up the cache
-		c.get()
-		cb.mockClear()
+
 		emitter.next(now())
-		expect(cb).toHaveBeenCalledTimes(1)
-	})
-	it('correctly caches undefined value', () => {
```
Comment (raveclassic) [inline packages/core/src/property.spec.ts:212]: This test should be replicated for the interop observable

Comment (Fyzu) [reply packages/core/src/property.spec.ts:212]: Moved to interop observable

**Review by raveclassic — APPROVED** [review]

**Pull request merged**